sdks/java/io/mqtt/src/test/java/org/apache/beam/sdk/io/mqtt/MqttIOTest.java

```diff
@@ -193,6 +195,83 @@ public void run() {
     publisherThread.join();
   }
 
+  /**
+   * Test for BEAM-3282: this test should not timeout.
+   */
+  @Test(timeout = 5 * 1000)
+  public void testReceiveWithTimeoutAndNoData() throws Exception {
+    pipeline.apply(MqttIO.read()
+        .withConnectionConfiguration(
+            MqttIO.ConnectionConfiguration.create(
+                "tcp://localhost:" + port,
+                "READ_TOPIC",
+                "READ_PIPELINE")).withMaxReadTime(Duration.standardSeconds(2)));
+
+    // should stop before the test timeout
+    pipeline.run();
+  }
+
+  /**
+   * Test for BEAM-3282: this test should not timeout.
+   */
+  @Test(timeout = 5 * 1000)
+  public void testReceiveWithTimeoutAndData() throws Exception {
```

Comment:
Hm this is a lot of copy-paste: maybe just change the other test that used withMaxNumRecords() to use withMaxReadTime() instead?

Reply:
Good idea, I'm doing that, rebase and squash. I will merge after then. Thanks !